components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -385,9 +385,14 @@ public static String getErrorPageURL(HttpServletRequest request, String errorCod
             return getErrorPageURL(request, errorCode, errorMessage, appName);
         } else {
             String redirectUri = request.getParameter(OAuthConstants.OAuth20Params.REDIRECT_URI);
+            String state = request.getParameter(OAuthConstants.OAuth20Params.STATE);
             try {
-                redirectUri += "?" + OAuthConstants.OAUTH_ERROR_CODE + "=" + URLEncoder.encode(errorCode, "UTF-8") +
-                        "&" + OAuthConstants.OAUTH_ERROR_MESSAGE + "=" + URLEncoder.encode(errorMessage, "UTF-8");
+                redirectUri += "?" + PROP_ERROR + "=" + URLEncoder.encode(errorCode, "UTF-8") +
```

Comment:
The callback URL can contain query params already

Reply:
@JKAUSHALYA let's use https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java#L1587 to build the URL

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1188